minpy/nn/solver.py

```diff
@@ -243,11 +270,16 @@ def train(self):
             start = time.time()
             self.epoch = epoch + 1
             for each_batch in self.train_dataiter:
-                self._step(each_batch)
+                self._step(each_batch, t + 1)
                 # Maybe print training loss
                 if self.verbose and t % self.print_every == 0:
                     print('(Iteration %d / %d) loss: %f' %
                           (t + 1, num_iterations, self.loss_history[-1]))
+                if self.visualize:
+                    # Add scalar summaries of training loss.
+                    loss_summary = summaryOps.scalarSummary('loss', self.loss_history[-1])
+                    self.train_writer.add_summary(loss_summary, t + 1)
```

Comment:
Since things are indexed from 0, I thought if this should start from 1. I can change back to t if you want.